src/main/java/org/openstreetmap/atlas/checks/validation/intersections/BoundaryIntersectionCheck.java

```diff
@@ -0,0 +1,403 @@
+    {
+        final Set<Relation> relations = currentLineItem.relations().stream()
+                .filter(relation -> relation instanceof MultiRelation).map(AtlasEntity::relations)
+                .flatMap(Collection::stream).collect(Collectors.toSet());
+        relations.addAll(currentLineItem.relations());
+        return relations.stream()
+                .filter(BoundaryIntersectionCheck::isRelationTypeBoundaryWithBoundaryTag)
+                .collect(Collectors.toSet());
+    }
+
+    private Set<Relation> getBoundaries(final Area area)
+    {
+        final Set<Relation> relations = area.relations().stream()
+                .filter(relation -> relation instanceof MultiRelation).map(AtlasEntity::relations)
+                .flatMap(Collection::stream).collect(Collectors.toSet());
+        relations.addAll(area.relations());
+        return relations.stream()
+                .filter(BoundaryIntersectionCheck::isRelationTypeBoundaryWithBoundaryTag)
+                .collect(Collectors.toSet());
+    }
+
+    private Set<BoundaryPart> getBoundaryParts(final Relation relation)
+    {
+        final RelationMemberList relationMemberLineItems = relation.membersOfType(ItemType.EDGE,
+                ItemType.LINE, ItemType.AREA);
+        return relationMemberLineItems.stream().map(RelationMember::getEntity).map(entity ->
+        {
+            final String tag = entity.getTags().get(BOUNDARY);
+            final Set<String> boundaryTags = entity.relations().stream()
+                    .filter(currentRelation -> BOUNDARY.equals(currentRelation.getTag(TYPE).get()))
+                    .map(currentRelation -> currentRelation.getTag(BOUNDARY)
+                            .orElse(StringUtils.EMPTY))
+                    .collect(Collectors.toSet());
+            boundaryTags.add(tag);
+            boundaryTags.remove(StringUtils.EMPTY);
+            return new BoundaryPart(entity, boundaryTags);
+        }).collect(Collectors.toSet());
+    }
+
+    private Geometry getGeometryForIntersection(final String wktFirst) throws ParseException
+    {
+
+        final WKTReader wktReader = new WKTReader();
+        Geometry geometry1 = wktReader.read(wktFirst);
+        if (geometry1.getGeometryType().equals(Geometry.TYPENAME_POLYGON))
+        {
+            geometry1 = geometry1.getBoundary();
+        }
+        return geometry1;
+    }
+
+    private Coordinate[] getIntersectionPoints(final String wktFirst, final String wktSecond)
+    {
+        try
+        {
+            final Geometry geometry1 = this.getGeometryForIntersection(wktFirst);
+            final Geometry geometry2 = this.getGeometryForIntersection(wktSecond);
+            return geometry1.intersection(geometry2).getCoordinates();
+        }
+        catch (final ParseException e)
+        {
+            throw new IllegalStateException(e);
+        }
+    }
+
+    private List<LineItem> getLineItems(final LineItem lineItem)
+    {
+        if (lineItem instanceof MultiLine)
```

Comment:
So i can process every LineItem that is part of MultiLine.

Reply:
I think there is some confusion here as to what a `MultiLine` or `MultiRelation` is in `Atlas`. These are not entities with an extra dimension of geometry. They are the types of entities stored in a `MultiAtlas`. For the most part they are functionally identical to their sibling types for in a `PackedAtlas` or `DynamicAtlas` (`PackedLine`, `DynamicRelation`, etc). These different types of `Atlas` implementations are used to load the `Atlas` in different ways, but should generally be treated the same once loaded. So for the different entity types you should just use the parent classes of `Area`, `Edge`, `Line`, `Node`, `Point`, and `Relation`. 
It would be extremely rare that you need to check for specific Packed, Multi, or Dynamic entity in an Atlas Check. In this case your check will only work if it is run on a `MultiAtlas`. This is not the default for the legacy or sharded checks spark jobs.

Reply:
@Bentleysb Thanks for the clarification, there was a confusion here. I will remove the obsolete code.